src/error.rs

```diff
@@ -584,3 +584,26 @@ pub(crate) fn convert_bulk_errors(error: Error) -> Error {
         _ => error,
     }
 }
+
+/// Flag a load-balanced mode mismatch.  In tests, it will panic; outside of tests, it will
```

Comment:
I like the idea of conditionally panicking, but what if we gated it on debug vs release (i.e. `#[cfg(debug_assertions)]`)? That way user test harnesses that may not be using cargo's test framework will also leverage this.

Also, what about returning in this macro instead of just evaluating to an expression? This would match the `todo!()` and `unimplemented!()` macros and would avoid accidentally progressing through an inconsistent state.

Reply:
Done.